### ISSUE TITLE
How can I protect $nuxt in console ?

### ISSUE THREAD

**Issue by achen224** [nuxt/nuxt.js#4720]
### What problem does this feature solve?

$nuxt should not show in console

### What does the proposed changes look like?

give $nuxt a scope and user can not use $nuxt by console

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8436">#c8436</a>)</em></sub></div>

**Comment by pi0** [comment]
Hi. You can use [globalName](https://nuxtjs.org/api/configuration-global-name#the-globalname-property) property to change the reference to a random secret one.

**Comment by achen224** [comment]
Thx @pi0 ,
But use globalName user still can call all fucntion by console

**Comment by pi0** [comment]
Even by not exposing a reference to the `window` object one can use other tricks like getting `vue` reference from DOM elements to access `$nuxt` (If not changed). Security is all about hardening. Assigning a random name would be the best solution:

```js
{
  globalName: Buffer.from(Math.random() + '').toString('base64').substr(0, 5)
}
```

**Comment by achen224** [comment]
ok, I just think it's too easy to access

![image](https://user-images.githubusercontent.com/10137830/50894200-9365e580-143d-11e9-9eeb-2998033471f8.png)

Thx for reply

**Comment by AndrewBogdanovTSS** [comment]
@Atinux yes, I also can't grasp what is the real issue here? Since JS is always fully exposed there is no real benefit in trying to hide some parts of it. @achen224 I think you should protect your server better, not the client

**Comment by paulvonber** [comment]
@manniL I think you're not really realising what you talking about... yes it's very secure to expose things like env variables.

**Comment by manniL** [comment]
@paulvonber No worries, I know **exactly** what I’m talking about 😁

> yes it's very secure to expose things like env variables.

This is a whole other topic and not related to that issue. If you somehow expose env variables accidentally, they are available in the JS source and therefore in the client anyway. With or without $nuxt, one could access them 🤷🏼‍♂️

**Comment by paulvonber** [comment]
Probably nuxt team accidentally exposing them as well. 
![image](https://user-images.githubusercontent.com/37245111/66584447-781f3680-eb85-11e9-9fa7-c69aa33b8398.png)
I notice you don't see any difference between typing $nuxt and trying to dig into js with debugger, 
According to you security doesn't exist and you can just print stuff on your home page because anyway everyone can find whatever they want.

**Comment by manniL** [comment]
Oh no 🙉 You’ve discovered the *public* Algolia API key!

I’d rather stop answering now as ad hominem arguments are not worth my time ☺️

**Comment by paulvonber** [comment]
@manniL your answers and arguments tells a lot about your knowledge. And whatever it is Algolia or anything else the fact is that the stuff is visible and it’s a hole.

**Comment by manniL** [comment]
> your answers and arguments tells a lot about your knowledge

I think so too 😋
Especially because I know what a difference between a public and a private key is 😂

But back to the topic:
1. Do your research, because from what I can read you are misinformed a little ☺️
2. I don’t like to repeat myself, but just for you: this is not a security issue, while showing private API keys would be (but that’s not what the issue is about after all :) )
3. If you have well-researched arguments, go ahead and open an issue with your thoughts.
4. If you eventually don’t like the decision, then feel free to fork Nuxt and do whatever you like
